### PR TITLE
new post: Trying new tools for building and automate the deployment in Kubernetes

### PR DESCRIPTION
**Pull request by kvaps**
This post provides an overview of my favorite tools to perform deploy to Kubernetes and also describes some useful practices to build your own pipeline with dynamic environments feature.

### CODE REVIEW

**Comment by onlydole** [comment]
Thank you for this post, @kvaps! I'll review this weekend and add feedback. 🙌🏼

**Comment by sftim** [comment]
I've spotted many, _small_ details that I think might be worth correcting. I will try to provide a review; at the moment, the draft feedback is hitting a GitHub internal size limit.

@kvaps please _don't_ be alarmed. Most of the things I'd be suggesting are really small things and I plan to provide GitHub suggestions that you can accept if you agree.

**Comment by kvaps** [comment]
@sftim sure, I'm not native english speaker, and I really value your help with that.
I look forward to your corrections, thank you!

File: content/en/blog/_posts/2020-01-11-try-new-deployment-tools.md [hunk 1]
```diff
@@ -0,0 +1,1232 @@
...
+   
+[**7.  Getting started with git-crypt**](#git-crypt)
+   
+[**8.  Preparing toolbox image**](#toolbox)
+   
+[**9.  Our first pipeline and building images using tags**](#pipeline-build)
+   
+[**10. Deployment automation**](#pipeline-deploy)
+   
+[**11. Artifacts and building on push to master**](#artefacts)
```
Comment (php-coder) [inline content/en/blog/_posts/2020-01-11-try-new-deployment-tools.md]:
```suggestion
[**11. Artifacts and building on push to master**](#artifacts)
```

File: content/en/blog/_posts/2020-01-11-try-new-deployment-tools.md [hunk 2]
```diff
@@ -0,0 +1,1232 @@
...
+
+And after the `git push` we will see how our applications were deployed:
+
+>
+> **[Screenshot of second pipeline](https://habrastorage.org/webt/0p/aj/vs/0pajvs-a-lrxvfuw8zylnj6lleg.png)**
+>
+
+---
+
+# 11. Artifacts and building on push to master {#artefacts}
```
Comment (php-coder) [inline content/en/blog/_posts/2020-01-11-try-new-deployment-tools.md]:
```suggestion
# 11. Artifacts and building on push to master {#artifacts}
```

File: content/en/blog/_posts/2020-01-11-try-new-deployment-tools.md [hunk 3]
```diff
@@ -0,0 +1,1232 @@
...
+
+>
+> **[Screenshot of docker-registry](https://habrastorage.org/webt/0-/31/qx/0-31qxfta2gk7-colutf2l_qtmm.png)**
+>
+
+---
+
+# 4. Getting started with QBEC {#qbec}
+
+Qbec is a deployment tool that allows you to declaratively describe the manifests of your application and deploy them to Kubernetes. Using Jsonnet as the main syntax allow to simplify the description of differences for several environments, and also almost completely eliminates code repeatability.
```
Comment (php-coder) [inline content/en/blog/_posts/2020-01-11-try-new-deployment-tools.md]: How about adding a link to Qbec official site?
```suggestion
[Qbec](https://qbec.io/) is a deployment tool that allows you to declaratively describe the manifests of your application and deploy them to Kubernetes. Using Jsonnet as the main syntax allow to simplify the description of differences for several environments, and also almost completely eliminates code repeatability.
```

File: content/en/blog/_posts/2020-01-11-try-new-deployment-tools.md [hunk 4]
```diff
@@ -0,0 +1,1232 @@
...
+
+If you have a pgp-key, then you can immediately add yourself as a collaborator for this project:
+
+```
+git-crypt add-gpg-user kvapss@gmail.com
+```
+
+Thus, you can always decrypt this repository using your private key.
+
+If you don’t have a pgp-key and do not plan to have it, then you can go the other way and export the project key:
```
Comment (php-coder) [inline content/en/blog/_posts/2020-01-11-try-new-deployment-tools.md]:
```suggestion
If you don’t have a PGP-key and do not plan to have it, then you can go the other way and export the project key:
```
?

**Comment by sftim** [comment]
Generally, I recommend you write command line parameters and paths like `--use-backticks`, and not as **\-\-bold\-text**

(If the stylesheet isn't quite write for the blog, lets fix that with CSS instead).

**Comment by kvaps** [comment]
Sure, I can use **`both`**

**Comment by kvaps** [comment]
@sftim, I've updated formatting to use backticks in https://github.com/kubernetes/website/pull/18630/commits/ac8ac93880c739f61a845a8c91e3c17964255001 now it looks much better

**Comment by onlydole** [comment]
@kvaps - can you please swap out the image in this post? I'd recommend an image that's more inclusive or relevant to the post. The current image comes across as insensitive.

Also, this post is a bit long and I’d recommend breaking it up into multiple posts and focusing on the narrative a bit more (ex. ”ok, now the nest step is foo, we do foo because it needs to do baz"). Before the table of contents in your post is a great start! The community likes posts that tell a story. 

I'll close out this PR in the meantime and we can evaluate any new PRs that you submit.

**Comment by kvaps** [comment]
@onlydole,
Image - ok

About the splitting the post, I'm not sure that it is good idea due:
1) The entire post is a separate story for the specific purpose of creating a statically generated site with dynamic environments using Kubernetes and my favorite tools. This post is more likely a tutorial, it describes all the steps to achieve the ultimate goal, along the way is talking about the tools used. This shows both use case and detailed review for each tool.
2) I'm not sure that separated parts will be relevant to Kubernetes itself, eg. Hugo and Gitlab configuration are not much relevant to it.

I already realized my fault that this post become quite long, but I do not intend to change it much.
If you refuse it, better I'll publish it on another place.

**Comment by sftim** [comment]
@kvaps, if you'd be willing, I can take the work you did and propose tweaks based on @onlydole's feedback.
Does that sound like something you'd be interested in?

**Comment by kvaps** [comment]
@sftim,
Thank you for your suggestion, I'm glad for any help, but what exactly do you would like to change?
How do you want to split it to many parts?

**Comment by kvaps** [comment]
Post is published on Medium:
https://medium.com/@kvaps/trying-new-tools-for-building-and-automate-the-deployment-in-kubernetes-f96f9684e580#83a1

**Pull request closed without merging**